neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -128,32 +131,58 @@ class NeuralynxRawIO(BaseRawIO):
         ("channel_id", "uint32"),
         ("sample_rate", "uint32"),
         ("nb_valid", "uint32"),
-        ("samples", "int16", (NcsSection._RECORD_SIZE)),
+        ("samples", "int16", NcsSection._RECORD_SIZE),
     ]
 
     def __init__(
-        self, dirname="", filename="", exclude_filename=None, keep_original_times=False, strict_gap_mode=True, **kargs
+            self,
+            dirname="",
+            include_filenames=None,
+            exclude_filenames=None,
+            keep_original_times=False,
+            strict_gap_mode=True,
+            **kargs
     ):
 
-        if dirname != "":
-            self.dirname = dirname
-            self.rawmode = "one-dir"
-        elif filename != "":
-            self.filename = filename
-            self.rawmode = "one-file"
+        if include_filenames is None:
+            include_filenames = []
+        elif not isinstance(include_filenames, (list, set, np.ndarray)):
+            include_filenames = [include_filenames]
+
+        if exclude_filenames is None:
+            exclude_filenames = set()
+        elif not isinstance(exclude_filenames, (list, set, np.ndarray)):
+            exclude_filenames = {exclude_filenames}
+
+        if include_filenames:
+            include_filepath = {os.path.dirname(f) for f in include_filenames}
+            if len(include_filepath) > 1:
+                raise ValueError("Files in include_filename must be in a single path!")
+
+        if (include_filenames is None) and (dirname == ""):
+            raise ValueError("One of dirname or include_filenames must be provided.")
```

Comment:
Basically the flow should be:

Check for deprecated arguments `filename` and `include_filename`: warn user about deprecation and explain that we now only accept dirname and that instead of include_filename we use include_filenames

 check for dirname. If dirname doesn't exist raise an error.

Then we check for include and exclude (those are optional).